benchmarktests/target_sync_aws.go

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package benchmarktests
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"math/rand"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/vault/api"
+	"github.com/mitchellh/mapstructure"
+	vegeta "github.com/tsenart/vegeta/v12/lib"
+)
+
+const (
+	SyncEvents            = "events"
+	SyncAssociationsWrite = "associations_write"
+	SyncAssociationsRead  = "associations_read"
+
+	secretNameFormat     = "my-benchmark-secret-%d"
+	vaultTokenHeader     = "X-Vault-Token"
+	vaultNamespaceHeader = "X-Vault-Namespace"
+)
+
+func init() {
+	TestList[SyncEvents] = func() BenchmarkBuilder {
+		return &SyncAWSTest{
+			target: SyncEvents,
+		}
+	}
+	TestList[SyncAssociationsWrite] = func() BenchmarkBuilder {
+		return &SyncAWSTest{
+			target: SyncAssociationsWrite,
+		}
+	}
+	TestList[SyncAssociationsRead] = func() BenchmarkBuilder {
+		return &SyncAWSTest{
+			target: SyncAssociationsRead,
+		}
+	}
+}
+
+type SyncAWSTest struct {
+	target string
+	mount  string
+
+	config *SyncAWSTestConfig
+
+	logger hclog.Logger
+}
+
+type SyncAWSTestConfig struct {
+	NumAssociations   int               `hcl:"num_associations,optional"`
+	DestinationType   string            `hcl:"destination_type"`
+	DestinationName   string            `hcl:"destination_name"`
+	DestinationConfig map[string]string `hcl:"destination_config,optional"`
+}
+
+func (t *SyncAWSTest) ParseConfig(body hcl.Body) error {
+	cfg := &struct {
+		Config *SyncAWSTestConfig `hcl:"config,block"`
+	}{
+		// Defaults
+		Config: &SyncAWSTestConfig{
+			NumAssociations:   1,
+			DestinationConfig: map[string]string{},
+		},
+	}
+
+	diags := gohcl.DecodeBody(body, nil, cfg)
+	if diags.HasErrors() {
+		return fmt.Errorf("error decoding to struct: %v", diags)
+	}
+
+	t.config = cfg.Config
+
+	return nil
+}
+
+func (t *SyncAWSTest) Flags(_ *flag.FlagSet) {}
+
+func (t *SyncAWSTest) Setup(client *api.Client, mountName string, topLevelConfig *TopLevelTargetConfig) (BenchmarkBuilder, error) {
+	t.logger = targetLogger.Named(t.target)
+
+	// Create test mount
+	t.logger.Debug(mountLogMessage("secrets", "kvv2", mountName))
+
+	if topLevelConfig.RandomMounts {
+		mountName += "-" + uuid.New().String()
+	}
```

Comment:
```suggestion
	if topLevelConfig.RandomMounts {
		mountName += "-" + uuid.New().String()
	}
	
	t.logger.Debug(mountLogMessage("secrets", "kvv2", mountName))
```
just swapped the order